scripts/environment_json.mjs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {getVersion} from "./get_version.mjs";
+import {getBuildNumber} from "./get_build_number.mjs";
+import url from "url";
+
+/*
+  Inputs:
+  => platform: the platform to generate the environment.json for
+  => release: whether or not this is a releasable binary
+
+  Outputs:
+  => the environment.json contents
+*/
+export async function environmentJson(platform, release) {
```

Comment:
Let's rename `release` to `buildType` and take one of debug or release. The argument should be mandatory. In the main() we can specify debug (or release?) as the default.

Maybe could maybe append "-debug" to the version in debug build types so we can see in the UI.

=== Side discussion ===
Maybe we can look into language that is used. Android uses language like "build variants", "build type", "product flavor": https://developer.android.com/studio/build/build-variants

Xcode seems to use "build configuration" for debug and release, but also has the notion of "Build Schemes"

Flutter seems to use the term "flavors": https://docs.flutter.dev/deployment/flavors.

Webpack has "[mode](https://webpack.js.org/configuration/mode/)", but also "[target](https://webpack.js.org/configuration/target/)"

In the end it's just a bunch of dimensions that direct how the app should be built.

Reply:
webpack's terminology has always rubbed me a bit the wrong way.

Of the remaining - `buildFlavor` seems to overlap solidly with a lot of these. However, internally we seem to use "binary" more that build - so "binaryFlavor," perhaps? It makes me think of ice cream 😋

Reply:
I'm just calling it flavor for now, will find-replace when we make a decision.